frameworks/hdfs/tests/test_overlay.py

```diff
@@ -58,8 +58,8 @@ def test_endpoints_on_overlay():
     expected_endpoints = ("hdfs-site.xml", "core-site.xml")
```

Comment:
I know this wasn't part of your changes, but this should ideally be a list and not a tuple.

Reply:
What is the problem with having this immutable?

Reply:
It's not a question of immutability, but rather that tuples are prone to errors when removing items. In this case, if we remove one of the elements then we'll end up iterating over the individual characters of the remaining string instead. 

This isn't critical as part of these changes though, and we can leave it as is.

Reply:
Is there even another way of removing an element from a tuple than casting to `list`, remove and cast back?

Reply:
I think I see now the struggle: In the future, it might happen that someone changes the initialization to `("hdfs-site.xml")` and misses the comma at the end... So the expression is evaluated to a string.